internal/staticanalysis/parsing/input_strategy.go

```diff
@@ -0,0 +1,82 @@
+package parsing
```

Comment:
This looks great!

Are there any unit-tests that can be added for this code?

Reply:
I will add some in a new PR